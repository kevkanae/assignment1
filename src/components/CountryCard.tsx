import {
  Box,
  Button,
  Card,
  CardActions,
  CardContent,
  Typography,
} from "@mui/material";
import { ICountryDetails } from "../interfaces/CountryDetails";
import { buttonStyles } from "../styles/ButtonStyles";
import { flexCol } from "../styles/flex";

interface ICountryCardProps {
  data: ICountryDetails;
  getWeatherSignal: Function;
}

const CountryCard = (props: ICountryCardProps) => {
  return (
    <Card
      sx={{
        width: "35vw",
        backgroundColor: "#14253d",
        borderRadius: 3,
        p: "1rem",
        color: "#f2ffff",
      }}
      elevation={7}
    >
      <CardContent>
        <Box
          component="img"
          sx={{
            height: "30vh",
            width: "100%",
            objectFit: "contain",
            mb: "1rem",
          }}
          alt="Flag"
          src={props.data.flags.svg}
        />
        <Box
          sx={{
            ...flexCol,
            justifyContent: "space-evenly",
          }}
        >
          <Typography
            sx={{
              fontSize: 21,
              fontWeight: 600,
              color: "white",
              textAlign: "center",
            }}
            gutterBottom
          >
            {props.data.name.official}
          </Typography>
          <Typography sx={{ fontSize: 18 }} gutterBottom>
            Population: {props.data.population}
          </Typography>
          <Typography sx={{ fontSize: 18 }} gutterBottom>
            LatLang: {props.data.latlng[0]}, {props.data.latlng[1]}
          </Typography>
          <Typography sx={{ fontSize: 18 }} gutterBottom>
            <a href={props.data.coatOfArms.png} style={{ color: "whitesmoke" }}>
              Image URL
            </a>
          </Typography>
        </Box>
      </CardContent>
      <CardActions>
        <Box
          sx={{
            ...flexCol,
            justifyContent: "center",
            p: "1.4rem",
            height: "10vh",
            width: "42vw",
          }}
        >
          <Button
            size="medium"
            sx={{
              ...buttonStyles,
            }}
            onClick={() => props.getWeatherSignal()}
          >
            Get Weather
          </Button>
        </Box>
      </CardActions>
    </Card>
  );
};

export default CountryCard;
