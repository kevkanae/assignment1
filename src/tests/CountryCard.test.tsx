import renderer from "react-test-renderer";
import "@testing-library/jest-dom/extend-expect";
import { screen, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import CountryCard from "../components/CountryCard";
import { QueryClient, QueryClientProvider } from "react-query";

const queryClient = new QueryClient();

const data = {
  name: {
    official: "",
  },
  capital: "",
  latlng: [1, 1],
  population: 111,
  flags: {
    png: "",
  },
  coatOfArms: {
    png: "",
  },
};

test("Snapshot of Country Card", () => {
  const comp = renderer.create(
    <BrowserRouter>
      <QueryClientProvider client={queryClient} contextSharing={true}>
        <CountryCard data={data} getWeatherSignal={() => {}} />
      </QueryClientProvider>
    </BrowserRouter>
  );
  let tree = comp.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("In Country Card,", () => {
  test("is the header present", () => {
    render(
      <BrowserRouter>
        <QueryClientProvider client={queryClient} contextSharing={true}>
          <CountryCard data={data} getWeatherSignal={() => {}} />
        </QueryClientProvider>
      </BrowserRouter>
    );

    const flagbox = screen.getByRole("flagbox");
    const offname = screen.getByRole("offname");
    const popu = screen.getByRole("popu");
    const cap = screen.getByRole("cap");
    const latlang = screen.getByRole("latlang");
    const urltoimg = screen.getByRole("urltoimg");
    const btn = screen.getByRole("weatherbtn");
    expect(flagbox).toBeInTheDocument();
    expect(offname).toBeInTheDocument();
    expect(popu).toBeInTheDocument();
    expect(cap).toBeInTheDocument();
    expect(latlang).toBeInTheDocument();
    expect(urltoimg).toBeInTheDocument();
    expect(btn).toBeInTheDocument();
  });
});
