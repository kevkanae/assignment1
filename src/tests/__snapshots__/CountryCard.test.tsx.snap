// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot of Country Card 1`] = `
<div
  className="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation7 MuiCard-root css-17prhnh-MuiPaper-root-MuiCard-root"
>
  <div
    className="MuiCardContent-root css-46bh2p-MuiCardContent-root"
  >
    <img
      alt="Flag"
      className="MuiBox-root css-l0b5fg"
      role="flagbox"
      src=""
    />
    <div
      className="MuiBox-root css-jwp3xc"
    >
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-1iuzxjv-MuiTypography-root"
        role="offname"
      />
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-15ejcyd-MuiTypography-root"
        role="popu"
      >
        Population: 
        111
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-15ejcyd-MuiTypography-root"
        role="cap"
      >
        Capital: 
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-15ejcyd-MuiTypography-root"
        role="latlang"
      >
        LatLang: 
        1
        ,
        1
      </p>
      <p
        className="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom css-15ejcyd-MuiTypography-root"
        role="urltoimg"
      >
        <a
          href=""
          style={
            Object {
              "color": "whitesmoke",
            }
          }
        />
      </p>
    </div>
  </div>
  <div
    className="MuiCardActions-root MuiCardActions-spacing css-1t6e9jv-MuiCardActions-root"
  >
    <div
      className="MuiBox-root css-17ptm9e"
    >
      <button
        className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-ko1l4s-MuiButtonBase-root-MuiButton-root"
        disabled={false}
        onBlur={[Function]}
        onClick={[Function]}
        onContextMenu={[Function]}
        onDragLeave={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        onKeyUp={[Function]}
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        role="weatherbtn"
        tabIndex={0}
        type="button"
      >
        Get Weather
      </button>
    </div>
  </div>
</div>
`;
